File: bfs.py
def bfs(matrix, cell):
    queue = []
    seen = []

    queue.append(cell)
    seen.append(cell)

    while (queue is not empty):
        current = queue.pop(0);

        # get the cells in 4 directions (not 9)
        surrounding_cells: int[] = get_surrounding_cells(current);

        for surrounding_cell in surrounding_cells:
            if cell == EXIT:
                return True
            elif (cell not in seen) and (cell != WALL):
                queue.append(surrounding_cell)
                seen.append(surrounding_cell)

File: bfs.c
#include "../libft/inc/libft.h"

typedef struct s_position {
	int	x;
	int	y;
} t_position;

int	bfs(int **map, t_position *start)
{
	t_list		*queue;
	t_list		*seen;
	t_position	*current;
	t_position			surrounding_cells[5];
	int			i;

	lst_add_back(queue, start);
	lst_add_back(seen, start);

	while (lst_size(queue) != 0)
	{
		current = lst_pop(queue, 0);
		if (!current)
			break ;

		surrounding_cells = get_surrounding_cells(current);
		i = 0;
		while (surrounding_cells[i])
		{
			if map[][]
		}
	}
}
